### DISCUSSION TITLE
How can I analysis the remix like file based routing and generate for react router ?

### DISCUSSION THREAD

**Discussion by wenerme** [remix-run/react-router#11944] [Q&A] [answered]
I like the remix's file base routing, I used for my react-router based project, it's a console like app, I don't need the remix's ability, but the file structure is good for orgnize the a lots of routes/pages.

e.g. I have a folder like this

<img width="241" alt="image" src="https://github.com/user-attachments/assets/530b6342-e639-46d9-8d23-b2b958918426">

I hope I can reuse the remix's file based routing to analysis these files, and generate routing config, is that possible ?
I think this also helps migrate to remix when needed.

**Comment by sergiodxa** [comment, marked as answer]
If you have `@remix-run/dev` script you can use `npx remix routes --json`, this will give you a JSON with the route configuration, although it's not fully compatible with React Router you could transform it to what createBrowserRoute expects.

**Reply by wenerme** [reply]
Tried this, seems this needs to add a lot deps `@remix-run/{dev,node,react}`, seems the tool do not detect react-router router ? The output is not nested in children which I expected. Maybe I should look at https://github.com/kiliman/remix-flat-routes
